includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -182,19 +182,37 @@ public static function has_delayed_payment( WC_Subscription $subscription ) {
 	 * @return WC_Subscription|bool The WC subscription or false if it can't be found.
 	 */
 	public static function get_subscription_from_wcpay_subscription_id( string $wcpay_subscription_id ) {
-		global $wpdb;
-
-		$subscription_id = $wpdb->get_var(
-			$wpdb->prepare(
-				"SELECT post_id FROM {$wpdb->prefix}postmeta
-				WHERE meta_key = %s
-				AND meta_value = %s",
-				self::SUBSCRIPTION_ID_META_KEY,
-				$wcpay_subscription_id
-			)
-		);
+		$query_args = [
```

Comment:
Can we just use `wcs_get_subscriptions()` without having to do the hook/filter meta query approach?

With older versions of WC Subscriptions this will still work as `wcs_get_subscriptions()` used to use WP Query and in new versions of WC Subscriptions or WC Pay Subscriptions, the code will just use the new `wcs_get_orders_with_meta_query()` function.

Reply:
Yes we can. 👍 

I've updated it to use `wcs_get_subscriptions()` in [b517a2f](https://github.com/Automattic/woocommerce-payments/pull/5130/commits/b517a2f93967fb82cdb0a0fa803d0bf3fea98871) and [d56fcfd](https://github.com/Automattic/woocommerce-payments/pull/5130/commits/d56fcfdbbeff1e5f60cf4d23fa3f95f034e23451).